src/Common/test/Common.OData.ApiExplorer.Tests/OData/DefaultModelTypeBuilderTest.cs

```diff
@@ -409,6 +409,41 @@ public void substitute_should_resolve_types_that_reference_a_model_that_match_th
         substitutionType.Should().NotBeOfType<TypeBuilder>();
     }
 
+    [Fact]
+    public void substituted_type_should_have_valid_runtime_properties__issue1104()
+    {
+        // arrange
+        var modelBuilder = new ODataConventionModelBuilder();
+
+        var address = modelBuilder.EntitySet<Address>( nameof( Address ) ).EntityType;
+        address.Ignore( x => x.City ); // force substitution
+        var addressType = typeof( Address );
+
+        var context = NewContext( modelBuilder.GetEdmModel() );
+
+        // act
+        var substitutedType = addressType.SubstituteIfNecessary( context );
+
+        // assert
+        substitutedType.Should().NotBe( addressType );
+        substitutedType.GetRuntimeProperties().Should().HaveCount( 5 )
+#if NET452
```

Comment:
I presume this was a limitation of FluentAssertions?

Reply:
Yes, due to different features support from FluentAssertions (or different Versions installed here, don't remember) depending on Framework.

I assume you're going to move to a different dep due to license (I recommend AwesomeAssertions).